File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.py
def solution(my_string, s, e):
    answer = list(my_string)
    arr = []
    for i in range(s,e+1):
        arr.insert(0,answer[i])
    for i in range(len(arr)):
        answer[i+s] = arr[i]
    return ''.join(answer)

File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.java
import java.util.ArrayList;

class Solution {
    public String solution(String my_string, int s, int e) {
        String answer = "";
        char[] strings = my_string.toCharArray();
        ArrayList<Character> reverse = new ArrayList<>();
        for (int i = s; i <= e; i++) {
            reverse.add(0, strings[i]);
        }
        for (int i = 0; i < reverse.size(); i++) {
            strings[i + s] = reverse.get(i);
        }
        for (char c : strings) {
            answer += c;
        }
        return answer;
    }
}